doc/layer_source_object.md

```diff
@@ -0,0 +1,177 @@
+
+### getValues(_step_)
+
+--- | ---
+Description | Returns the list of values for the pixels active in `step`.
+Returns | List of values.
+
+### invalidate()
+
+--- | ---
+Description | Forces a reload of the layer data.
+
+#### Example
+
+```javascript
+ <script>
+// define the torque layer style using cartocss
+// this creates a kind of density map
+// color scale from http://colorbrewer2.org/
+var CARTOCSS = [
+  'Map {',
+  '-torque-time-attribute: "date";',
+  '-torque-aggregation-function: "avg(temp::float)";',
+  '-torque-frame-count: 1;',
+  '-torque-animation-duration: 15;',
+  '-torque-resolution: 16',
+  '}',
+  '#layer {',
+  '  marker-width: 8;',
+  '  marker-fill-opacity: 1.0;',
+  '  marker-fill: #fff5eb; ',
+  '  marker-type: rectangle;',
+  '  [value > 1] { marker-fill: #fee6ce; }',
+  '  [value > 2] { marker-fill: #fdd0a2; }',
+  '  [value > 4] { marker-fill: #fdae6b; }',
+  '  [value > 10] { marker-fill: #fd8d3c; }',
+  '  [value > 15] { marker-fill: #f16913; }',
+  '  [value > 20] { marker-fill: #d94801; }',
+  '  [value > 25] { marker-fill: #8c2d04; }',
+  '}'
+].join('\n');
+
+var map = new L.Map('map', {
+  zoomControl: true,
+  center: [40, 0],
+  zoom: 3
+});
+L.tileLayer('http://{s}.api.cartocdn.com/base-dark/{z}/{x}/{y}.png', {
+  attribution: 'CartoDB'
+}).addTo(map);
+var torqueLayer = new L.TorqueLayer({
+  user : 'viz2',
+  table : 'ow',
+  cartocss: CARTOCSS
+});
+torqueLayer.addTo(map);
+map.on('click', function(e) {
+  var p = e.containerPoint
+  var value = torqueLayer.getValueForPos(p.x, p.y);
+  if (value !== null) {
+    map.openPopup('average temperature: ' + value.value + "C", e.latlng);
+  }
+});
+```
+
+## Named Maps Layer Source Object (_type: 'namedmap'_)
+
+Used for making public maps with private data. See [Named Maps](/cartodb-platform/maps-api/named-maps/) for more information.
+
+#### Example
+
+```javascript
+{
+  user_name: 'your_user_name', // Required
+  type: 'namedmap', // Required
+  named_map: {
+    name: 'name_of_map', // Required
+    // Optional
+    layers: [{
+      layer_name: "sublayer0", // Optional
+      interactivity: "column1, column2, ..." // Optional
+    },
+    {
+      layer_name: "sublayer1",
+      interactivity: "column1, column2, ..."
+    },
+      ...
+    ],
+    // Optional
+    params: {
+      color: "hex_value",
+      num: 2
+    }
+  }
+}
+```
+
+### Multiple types of layers Source Object
+
+`cartodb.createLayer` combining multiple types of layers and setting a filter
+
+#### Example
+
+```javascript
+cartodb.createLayer(map, {
+  user_name: 'examples',
+  type: 'cartodb',
+  sublayers: [
+    {
+      type: "http",
+      urlTemplate: "http://{s}.basemaps.cartocdn.com/dark_nolabels/{z}/{x}/{y}.png",
+      subdomains: [ "a", "b", "c" ]
+    },
+    {
+     sql: 'select * from country_boundaries',
+     cartocss: '#layer { polygon-fill: #F00; polygon-opacity: 0.3; line-color: #F00; }'
+    },
+  ],
+}, { filter: ['http', 'mapnik'] })
+```
```

Comment:
Looks good to me! (But all the Torque related content will be removed once we merge the Torqe.js branch, correct?)

Just want to confirm, as this comment appears in the expose-torquejs WIP branch: {% comment %}writer note_csobier: 14-Dec-2015. Since these advanced interaction methods are specific to Torque, I deleted this dup content from the CartoDB.js/api_methods.md doc. This content now only appears here, in the Torque.js docs folder/03-torque_interaction_method.{% endcomment %}